app/controllers/sites/new.js

```diff
@@ -3,48 +3,117 @@ import { inject as service } from '@ember/service';
 import { combineFullAddress } from 'frontend-contactgegevens-loket/models/address';
 import { tracked } from '@glimmer/tracking';
 import { task } from 'ember-concurrency';
+import { action } from '@ember/object';
+import {
+  errorValidation,
+  warningValidation,
+} from '../../validations/site-validation';
 
 export default class CreateSitesNewController extends Controller {
   @service router;
   @service store;
   @tracked isPrimarySite = false;
+  @tracked validationErrors = {};
+  @tracked validationWarnings = {};
+  @tracked showWarningModal = false;
+
+  @action
+  reset() {
+    this.isPrimarySite = false;
+    this.validationErrors = {};
+    this.validationWarnings = {};
+    this.showWarningModal = false;
+  }
 
   get isLoading() {
     return this.saveTask.isRunning || this.cancelTask.isRunning;
   }
 
+  async validateData() {
+    const { address, primaryContact, secondaryContact, site } = this.model;
+    const validationData = {
+      siteType: site.siteType.get('label'),
+      street: address.street,
+      country: address.country,
+      number: address.number,
+      postcode: address.postcode,
+      municipality: address.municipality,
+      province: address.province,
+      fullAddress: address.fullAddress,
+      telephonePrimary: primaryContact.telephone
+        ? primaryContact.telephone.replace(/ /g, '')
+        : '',
+      emailPrimary: primaryContact.email,
+      websitePrimary: primaryContact.website,
+      telephoneSecondary: secondaryContact.telephone,
+    };
+
+    const errorValidationResult = errorValidation.validate(validationData);
+    const warningValidationResult = warningValidation.validate(validationData);
+    return {
+      errors: errorValidationResult.error,
+      warnings: warningValidationResult.error,
+    };
+  }
+
   saveTask = task(async (event) => {
     event.preventDefault();
-    const { address, primaryContact, secondaryContact, site, adminUnit } =
-      this.model;
-    address.fullAddress = combineFullAddress(address);
-    await primaryContact.save();
-    await secondaryContact.save();
-    await address.save();
-    site.contacts = [primaryContact, secondaryContact];
-    site.address = address;
-    await site.save();
-    const nonPrimarySites = await adminUnit.sites;
-
-    if (this.isPrimarySite) {
-      const previousPrimarySite = await adminUnit.primarySite;
-      if (previousPrimarySite) {
-        nonPrimarySites.push(previousPrimarySite);
-      }
-      adminUnit.primarySite = site;
-    } else {
-      nonPrimarySites.push(site);
+    this.validationErrors = {};
+    this.validationWarnings = {};
+    const validationResult = await this.validateData();
+    if (
+      validationResult.errors &&
+      Object.keys(validationResult.errors).length > 0
+    ) {
+      // Handle errors
+      this.validationErrors = validationResult.errors.details.reduce(
```

Comment:
Integrate this reducer operation into the validateData function. Now the reducer code is duplicated for errors and warnings.